crates/ubrn_bindgen/src/bindings/react_native/gen_typescript/templates/InitializationTemplate.ts

```diff
@@ -0,0 +1,19 @@
+(() => {
+    // Get the bindings contract version from our ComponentInterface
+    const bindingsContractVersion = {{ ci.uniffi_contract_version() }};
+    // Get the scaffolding contract version by calling the into the dylib
+    const scaffoldingContractVersion = nativeModule().{{ ci.ffi_uniffi_contract_version().name() }}();
+    if (bindingsContractVersion != scaffoldingContractVersion) {
+        throw new Error("contractVersionMismatch");
+    }
+
+    {%- for (name, expected_checksum) in ci.iter_checksums() %}
+    if (nativeModule().{{ name }}() != {{ expected_checksum }}) {
+        throw new Error("apiChecksumMismatch: {{ name }}");
+    }
+    {%- endfor %}
+
+    {% for fn in self.initialization_fns() -%}
+    {{ fn }}();
+    {% endfor -%}
```

Comment:
This was the real reason for this: at start-up, we register the callback vtable with C++ and then onwards to Rust. Only after that can user code use callbacks.